addons/l10n_ar_stock/models/stock_picking.py

```diff
@@ -0,0 +1,91 @@
+from base64 import b64encode
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    l10n_ar_delivery_guide_number = fields.Char(
+        string="Delivery Guide No.",
+        copy=False,
+        readonly=True,
+    )
+    l10n_ar_cai_data = fields.Json(string="CAI Data")
+    l10n_ar_allow_generate_delivery_guide = fields.Boolean(
+        compute="_compute_l10n_ar_delivery_guide_flags",
+    )
+    l10n_ar_allow_send_delivery_guide = fields.Boolean(
+        compute="_compute_l10n_ar_delivery_guide_flags",
+    )
+
+    # === Compute Methods ===
+
+    @api.depends('state', 'l10n_ar_delivery_guide_number', 'picking_type_id.l10n_ar_document_type_id')
+    def _compute_l10n_ar_delivery_guide_flags(self):
+        """
+        Compute flags for allowing delivery guide generation and sending.
+        - Generation allowed if: state is 'done', document type exists, and no guide number.
+        - Send allowed if: guide number exists.
+        """
+        for picking in self:
+            has_doc_type = bool(picking.picking_type_id.l10n_ar_document_type_id)
+            picking.l10n_ar_allow_generate_delivery_guide = (
+                picking.state == 'done' and has_doc_type and not picking.l10n_ar_delivery_guide_number
+            )
+            picking.l10n_ar_allow_send_delivery_guide = bool(picking.l10n_ar_delivery_guide_number)
+
+    # === Business Methods ===
+
+    def l10n_ar_action_create_delivery_guide(self):
+        """
+        Create the delivery guide number and store CAI data for the stock picking.
+        """
+        self.ensure_one()
+        self._check_company()
+
+        if not self.l10n_ar_delivery_guide_number:
+            picking_type = self.picking_type_id
+            self.l10n_ar_delivery_guide_number = f'{picking_type.l10n_ar_delivery_sequence_prefix}-{picking_type.l10n_ar_sequence_id.next_by_id()}'
```

Comment:
Can we instead set the `l10n_ar_delivery_sequence_prefix` as the `prefix` on the `ir.sequence`? That feels a bit more cohesive rather than manually concatenating them here. We can then make the prefix field on `stock.picking.type` a related to the sequence, just like you did for `l10n_ar_next_delivery_number`.

```suggestion
            self.l10n_ar_delivery_guide_number = picking_type.l10n_ar_sequence_id.next_by_id()
```